### PR TITLE
Show right code line when server crashes.

### PR DESCRIPTION
**Pull request by jakozaur**
On server crash, show the actual code line instead of showing always line from `Fiber` library.

Makes Meteor debugging more fun and beginner friendly!

Fixes:
https://github.com/meteor/meteor/issues/4598

### CODE REVIEW

**Comment by rclai** [comment]
+1

**Comment by joaopiopedreira** [comment]
+1

**Comment by glasser** [comment]
So is this basically just code borrowed from source-map-support with some tweaks?  What are the differences?  Can you include a test (in `tools/tests`) that shows this working correctly?  There's definitely some subtleties to this code that are hard to wrap my head around.

**Comment by glasser** [comment]
Or to be more clear --- I don't understand what about this change removes the fibers line specifically.  Or how it really removes anything... can you explain how it accomplishes its goal (preferably in comments)?

**Comment by jakozaur** [comment]
Thanks so much @glasser for the response. Looks like you wrote the most of Meteor code. Hats off!

There are two problems which cause showing the wrong line:
1. If you don't register `uncaughtException` then it always show the line from `Fiber`.
2. Even if set flag in source-map-support to do that, it shows line only from file when it has source map. In our case it always shows line from underscore, because regex matches:
```
at Function._each_.forEach (/some/path:123:123)
```
not the desired:
```
at app/server/publish.js:123:123
```

So the fix is to use regexes to look for both frames with or without source maps. Take one that is closer to top and show it.

Will try to add test. Also feel free to ask more questions.

Alternative solutions that I considered:
1. Generate source maps for all files. Don't know how to do that. Moreover, it may slow down the startup of server.
2. Fix that in [source-map-support](https://github.com/evanw/node-source-map-support/blob/master/source-map-support.js). However, I don't know how to work with development version of node package. Moreover, it is actual feature for them (they show line from your code, not some internal library) and we can't solve it easily in general case. In case of meteor we know that is files are not minified, in non-Meteor case it can end up in showing garbage. Moreover, we don't care about the browser, but they do.

Tl;dr: The fix is Meteor specific, not sure if we can make it universal for source-map-support.

**Comment by benjamn** [comment]
Thanks for this PR, sorry we never got around to merging it. We are currently cleaning up the repository and removing old PRs; if you think this PR still has value and are interested in resurrecting it, we encourage you to re-submit it against current `devel`.
